### PR TITLE
Add support for a separate interval on agent execution failure.

### PR DESCRIPTION
**Pull request by rfarrjr**
Fiat has migrated to leverage cats but needs the ability to have a separate agent interval when the agent execution fails.   This enhancement to cats supports this functionality while maintaining backwards compatibility.

### CODE REVIEW

File: cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/AgentIntervalProvider.java [hunk 1]
```diff
@@ -24,10 +24,16 @@
...
     public static class Interval {
         final long interval;
+        final long errorInterval;
         final long timeout;
 
         public Interval(long interval, long timeout) {
+            this(interval, interval, timeout);
+        }
+
+        public Interval(long interval, long errorInterval, long timeout) {
```
Comment (TrevinTeacutter) [inline cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/AgentIntervalProvider.java:34]: Very minor nit: Shouldn't the extra param be the last param for this overloaded constructor?

**Review by TrevinTeacutter — APPROVED** [review]

**Review by mtweten — APPROVED** [review]

File: cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/ClusteredAgentScheduler.java [hunk 2]
```diff
@@ -152,21 +157,28 @@ public void unschedule(Agent agent) {
...
     private static class AgentJob implements Runnable {
         private final long lockReleaseTime;
+        private final long lockErrorReleaseTime;
         private final AgentExecutionAction action;
         private final ClusteredAgentScheduler scheduler;
 
-        public AgentJob(long lockReleaseTime, AgentExecutionAction action, ClusteredAgentScheduler scheduler) {
-            this.lockReleaseTime = lockReleaseTime;
+        public AgentJob(Long[] times, AgentExecutionAction action, ClusteredAgentScheduler scheduler) {
+            this.lockReleaseTime = times[0] + times[1];
```
Comment (TrevinTeacutter) [inline cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/ClusteredAgentScheduler.java]: Maybe add another enum or const ints to give more context to what each index of times means without digging into the code further?

File: cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/AgentIntervalProvider.java (hunk 1 above)
Comment (mtweten) [reply cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/AgentIntervalProvider.java:34]: Think it makes sense to group the intervals together. It's a whole new constructor so there aren't any limitations on what can be changed.

Comment (rfarrjr) [reply cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/AgentIntervalProvider.java:34]: yeah, I debated that both ways.  Finally decided to group the intervals.

**Pull request merged**